prow/plugins/trigger/trigger.go

```diff
@@ -250,9 +212,15 @@ func RunOrSkipRequested(c Client, pr *github.PullRequest, requestedJobs []config
 	toRun := sets.NewString()
 	toSkip := sets.NewString()
 	for _, job := range requestedJobs {
-		runs, err := shouldRun(job)
-		if err != nil {
-			return err
+		// jobs that don't run on the branch in question are in the forced list as
+		// that is how branch sharding is implemented, so we need to do that check
+		// twice
+		runs := job.RunsAgainstBranch(pr.Base.Ref) && forceRunContexts.Has(job.Context)
+		if !runs {
+			runs, err = job.ShouldRun(pr.Base.Ref, body, config.NewGitHubDeferredChangedFilesProvider(c.GitHubClient, org, repo, number))
```

Comment:
I kind of feel like `ShouldRun()` should take `forceRunContexts` as a parameter here instead of doing the check twice, but if this is the only place this is needed (to support`/retest`) this makes more sense like this for now. /shrug